app/controllers/schedule_controller.rb

```diff
@@ -2,33 +2,38 @@
 
 class ScheduleController < ApplicationController
   def save_scheduled_publishing_datetime
-    Edition.find_and_lock_current(document: params[:document]) do |edition|
-      checker = Requirements::ScheduledDatetimeChecker.new(permitted_params)
-      issues = checker.pre_submit_issues
-
-      if issues.any?
-        flash["alert_with_items"] = {
-          title: I18n.t!("requirements.scheduled_datetime.title"),
-          items: issues.items(
-            link_options: {
-              scheduled_datetime: { href: "#scheduled_publishing_datetime" },
-            },
-          ),
-        }
-        flash[:scheduled_publishing_params] = permitted_params
-        redirect_to document_path(edition.document)
-        next
-      end
-
-      set_scheduled_publishing_datetime(edition, checker.parsed_datetime)
-
+    result = Schedule::SaveDatetimeInteractor.call(params: params, user: current_user)
+    edition, issues = result.to_h.values_at(:edition, :issues)
+
+    if issues
+      flash.now["alert_with_items"] = {
+        "title" => I18n.t!("requirements.scheduled_datetime.title"),
+        "items" => issues.items(
+          link_options: {
+            scheduled_datetime: { href: "#scheduled_publishing_datetime" },
+          },
+        ),
+      }
+
+      render :scheduled_publishing_datetime,
+             assigns: { edition: edition, issues: issues },
+             status: :unprocessable_entity
+    elsif params[:schedule][:action] == "schedule"
+      redirect_to scheduling_confirmation_path(edition.document)
+    else
       redirect_to document_path(edition.document)
     end
   end
 
   def clear_scheduled_publishing_datetime
     Edition.find_and_lock_current(document: params[:document]) do |edition|
-      set_scheduled_publishing_datetime(edition)
+      updater = Versioning::RevisionUpdater.new(edition.revision, current_user)
+      updater.assign(scheduled_publishing_datetime: nil)
+
+      if updater.changed?
+        edition.assign_revision(updater.next_revision, current_user).save!
+        create_datetime_cleared_timeline_entry(edition)
+      end
```

Comment:
Sure, I thought about doing this in this PR but we've got a card to review clearing so I think it can be done as part of that.  Will add as an A/C to that.

Reply:
Ok cool 👍